python/dgl/distributed/dist_graph.py

```diff
@@ -490,7 +497,12 @@ def rank(self):
         int
             The rank of the current graph store.
         '''
-        return self._client.client_id
+        if self._gpb is None:
+            return self._client.client_id
+        else:
+            num_client_per_part = self._num_client // self._gpb.num_partitions()
+            client_id_in_part = self._client.client_id % num_client_per_part
```

Comment:
@aksnzhy how is the client_id decided in kvstore client?

Reply:
The code is right. We assume that the client id is grouped together inside a machine.